### ISSUE TITLE
Ensure we don't use `yield_once` unnecessarily

### ISSUE THREAD

**Issue by Nosferatul** [coala/coala#3917]
https://github.com/coala/coala/issues/2912

**Comment by Makman2** [comment]
Some example?

**Comment by Nosferatul** [comment]
Not yet, sorry for the late response

**Comment by palash25** [comment]
@Nosferatul we only use `yield_once` 6 times in coalib in `Importers.py`, `Collectors.py` and `Globbing.py` can you give an example how this might be slowing performance?

**Comment by Nosferatul** [comment]
I used a program to check what functions might slow performance but maybe it was a mistake

**Comment by Makman2** [comment]
Yeah I also remember that it was quite slow because it was involved in our file processing. `yield_once` is anyway slightly special, if it makes sense to remove it, we should rather avoid it :+1:
